services/py-api/src/utils.py

```diff
@@ -1,7 +1,9 @@
+from os import environ
 from threading import Lock
-from typing import Any, Dict
-
+from typing import Any, Dict, TypedDict
+from jwt import DecodeError, ExpiredSignatureError, InvalidSignatureError, decode, encode
 import httpx
+import jwt
```

Comment:
There is no reason to import the whole jwt library again. Add whatever dependency you need from jwt in the import statement in line no.4

Reply:
In the decode method I use "jwt.decode" and without importing the whole library it writes it is not defined

Reply:
Also please install the dependecy using poetry. You have not added jwt in the dependencies. 

```
poetry add pyjwt
```

Reply:
do from jwt import jwt instead of import jwt

Reply:
> There is no reason to import the whole jwt library again. Add whatever dependency you need from jwt in the import statement in line no.4

To clarify, as this comment might seem a bit confusing: Both from `<package> import ...` and `import <package>` load the entire package into memory. However, there's a key difference in how you access the package's attributes.

With `import <package>`, you can access any attribute of the package by explicitly using <package>.<attribute>.

With `from <package> import ...`, the imported attributes are added directly to the current namespace. This allows you to access them without prefixing them with the package name. However, this approach can clutter the namespace, potentially leading to conflicts or ambiguity if multiple attributes with the same name are imported from different sources.

Having experienced both sides of this issue, I generally advise against using `from ... import ...`

Reply:
@asynchroza Thank you for the correction!

Reply:
Fixed in be223527926be992bf6729e80ee043728b8fa1c5